torchvision/models/detection/anchor_utils.py

```diff
@@ -132,27 +130,17 @@ def grid_anchors(self, grid_sizes: List[List[int]], strides: List[List[Tensor]])
 
         return anchors
 
-    def cached_grid_anchors(self, grid_sizes: List[List[int]], strides: List[List[Tensor]]) -> List[Tensor]:
-        key = str(grid_sizes) + str(strides)
-        if key in self._cache:
-            return self._cache[key]
-        anchors = self.grid_anchors(grid_sizes, strides)
-        self._cache[key] = anchors
-        return anchors
-
     def forward(self, image_list: ImageList, feature_maps: List[Tensor]) -> List[Tensor]:
         grid_sizes = list([feature_map.shape[-2:] for feature_map in feature_maps])
         image_size = image_list.tensors.shape[-2:]
         dtype, device = feature_maps[0].dtype, feature_maps[0].device
         strides = [[torch.tensor(image_size[0] // g[0], dtype=torch.int64, device=device),
                     torch.tensor(image_size[1] // g[1], dtype=torch.int64, device=device)] for g in grid_sizes]
         self.set_cell_anchors(dtype, device)
```

Comment:
I believe that the set_cell_anchors can also be refactored. The implementation currently stores a cached version of it as attribute and uses it as long as the device is the same. I think this is a bit awkward.

Instead we could move this estimation to the constructor (because ignoring dtype and device, it depends only on `sizes` and `aspect_ratios`) and copy/cast them to the current device/dtype during forward. This way we pre-estimate stuff and avoid those weird caches (and their assumptions). Happy if you want this to be done on a follow up PR.

Reply:
I can create a separate PR for that.